pkg/builder/builder_test.go

```diff
@@ -86,21 +100,107 @@ func TestPrepareBuild(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
+	checksPerformed := 0
 
-	if len(b.InvocationImages) != 1 {
-		t.Fatalf("expected there to be 1 image, got %d. Full output: %v", len(b.Images), b)
+	if !reflect.DeepEqual(b.Actions, mfst.Actions) {
+		t.Errorf("expected actions to be %+v but was %+v", mfst.Actions, b.Actions)
 	}
+	checksPerformed++
 
-	if b.Version != mfst.Version {
-		t.Errorf("expected version %v, got %v", mfst.Version, b.Version)
+	if !reflect.DeepEqual(b.Credentials, mfst.Credentials) {
+		t.Errorf("expected credentials to be %+v but was %+v", mfst.Credentials, b.Credentials)
 	}
-	if b.SchemaVersion != mfst.SchemaVersion {
-		t.Errorf("expected schemaVersion %v, got %v", mfst.SchemaVersion, b.SchemaVersion)
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Custom, mfst.Custom) {
+		t.Errorf("expected custom to be %+v but was %+v", mfst.Custom, b.Custom)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Definitions, mfst.Definitions) {
+		t.Errorf("expected definitions to be %+v but was %+v", mfst.Definitions, b.Definitions)
 	}
+	checksPerformed++
+
+	if b.Description != mfst.Description {
+		t.Errorf("expected description to be %+v but was %+v", mfst.Description, b.Description)
+	}
+	checksPerformed++
+
+	if len(b.InvocationImages) != 1 {
+		t.Fatalf("expected there to be 1 image, got %d. Full output: %v", len(b.Images), b)
+	}
+	checksPerformed++
+
 	expected := bundle.InvocationImage{}
 	expected.Image = "cnab:0.1.0"
 	expected.ImageType = "docker"
 	if !reflect.DeepEqual(b.InvocationImages[0], expected) {
 		t.Errorf("expected %v, got %v", expected, b.InvocationImages[0])
 	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Keywords, mfst.Keywords) {
+		t.Errorf("expected keywords to be %+v but was %+v", mfst.Keywords, b.Keywords)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Maintainers, mfst.Maintainers) {
+		t.Errorf("expected maintainers to be %+v but was %+v", mfst.Maintainers, b.Maintainers)
+	}
+	checksPerformed++
+
+	if b.Name != mfst.Name {
+		t.Errorf("expected name to be %+v but was %+v", mfst.Name, b.Name)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Outputs, mfst.Outputs) {
+		t.Errorf("expected outputs to be %+v but was %+v", mfst.Outputs, b.Outputs)
+	}
+	checksPerformed++
+
+	if !reflect.DeepEqual(b.Parameters, mfst.Parameters) {
+		t.Errorf("expected parameters to be %+v but was %+v", mfst.Parameters, b.Parameters)
+	}
+	checksPerformed++
+
+	if b.SchemaVersion != mfst.SchemaVersion {
+		t.Errorf("expected schemaVersion %v, got %v", mfst.SchemaVersion, b.SchemaVersion)
+	}
+	checksPerformed++
+
+	if b.Version != mfst.Version {
+		t.Errorf("expected version %v, got %v", mfst.Version, b.Version)
+	}
+	checksPerformed++
+
+	mfstFields := getFields(manifest.Manifest{})
+	if len(mfstFields) != checksPerformed {
+		t.Errorf("expected to check %v fields for equality, but checked only %v fields", len(mfstFields), checksPerformed)
+	}
+}
+
+func TestBundleAndManifestHasSameFields(t *testing.T) {
```

Comment:
I'm not convinced that it's worth checking that the field names of the structures are the same since the field names do not actually have to be the same and it seems a bit odd comparing the names of fields with distinct types. Given the previous test, I think we could safely delete this one.

Reply:
nit: if we decide to keep this test, the name should be changed:
```suggestion
func TestBundleAndManifestHaveSameFields(t *testing.T) {
```

Reply:
> I'm not convinced that it's worth checking that the field names of the structures are the same since the field names do not actually have to be the same and it seems a bit odd comparing the names of fields with distinct types. Given the previous test, I think we could safely delete this one.

In my previous PR: https://github.com/deislabs/duffle/pull/790 the SchemaVersion field was missing in the manifest, but yet elsewhere in duffle a check was performed to see if this field existed and was set. Hence this test. 

IMHO having both bundle and manifest is a bit redundant, there really should be only one struct, but as mentioned in the slack: https://cloud-native.slack.com/archives/CEX1W7WMD/p1562696449224400 InvocationImage is the only field that is different.

Reply:
I'm comfortable working on the assumption that the field names should be the same in both structures until such time as we need to change that. It _should_ be obvious that the test is overly restrictive.